Umweltzone/src/main/java/de/avpptr/umweltzone/zones/viewholders/OneZoneViewHolder.kt

```diff
@@ -32,19 +32,15 @@ class OneZoneViewHolder(
 
 ) : ZoneViewHolder<ZoneViewModel.OneZoneViewModel>(view) {
 
-    private val zoneShapeView: GradientDrawable
-
-    init {
-        val badgeBackground = view.zoneOneZoneBadgeView.background as LayerDrawable
-        zoneShapeView = badgeBackground.findDrawableByLayerId(R.id.zone_shape) as GradientDrawable
-    }
-
     override fun bind(viewModel: ZoneViewModel.OneZoneViewModel) = with(view) {
+        val binding = ZonesListItemOneZoneBinding.bind(view)
```

Comment:
You can get rid of the `view` and the `layout` when you pass the `binding` via the constructor instead.

``` kotlin
class OneZoneViewHolder(

        private val binding: ZonesListItemOneZoneBinding,
        private val onItemClick: (view: View) -> Unit

) : ZoneViewHolder<ZoneViewModel.OneZoneViewModel>(binding.root) { ... }
```

Call site:

``` kotlin
ChildZonesCount.ONE.value -> {
    val binding = ZonesListItemOneZoneBinding.inflate(inflater, parent, false)
    OneZoneViewHolder(binding, onItemClick)
}
```

Reply:
Done.

Reply:
@fejd Did you forget to push your changes?

Reply:
Thanks for the pointers to the generated bindings by the way. I had a look and confirmed your findings.